src/java/org/apache/cassandra/gms/FailureDetector.java

```diff
@@ -476,16 +470,12 @@ public long[] getArrivalIntervals()
 
     private static long getMaxInterval()
     {
-        String newvalue = System.getProperty("cassandra.fd_max_interval_ms");
```

Comment:
same logic as above

Reply:
I updated this part according to your suggestions in general.

The main thing here is that the `getMaxInterval` method uses the `INITIAL_VALUE_NANOS` constant as a default value. Which in turn is initialized by the `getInitialValue` method call and can't be changed in runtime further, thus changing the `cassandra.fd_initial_value_ms` will have no effect until restart.